### PR TITLE
Check for invalid argus instance error code in SKE creation waiter

### PR DESCRIPTION
**Pull request by vicentepinto98**

### CODE REVIEW

File: services/ske/wait.go [hunk 1]
```diff
@@ -50,11 +52,21 @@ func CreateOrUpdateClusterWaitHandler(ctx context.Context, a APIClientClusterInt
 			return handleError(err)
 		}
 		state := *s.Status.Aggregated
+
+		// The state "STATE_UNHEALTHY" (aka "Impaired" in the portal) could be temporarily occur during cluster creation and the system is recovering usually, so it is not considered as a failed state here.
+		// -- alignment meeting with SKE team on 4.8.23
+		// The exception is when providing an invalid argus instance id, in that case the cluster will stay as "Impaired" until the SKE team solves it, but it is still usable.
+		// In this caseand the waiter should succeed and in the terraform provider add a warning
```
Comment (hcsa73) [inline services/ske/wait.go:59]: Typo in this line, also comment regarding the TFP shouldn't be here

File: services/ske/wait.go [hunk 2]
```diff
@@ -50,11 +52,21 @@ func CreateOrUpdateClusterWaitHandler(ctx context.Context, a APIClientClusterInt
...
+		if state == stateUnhealthy {
+			if s.Status.Error != nil && s.Status.Error.Message != nil && *s.Status.Error.Code == InvalidArgusInstanceErrorCode {
```
Comment (hcsa73) [inline services/ske/wait.go:61]: I'd merge this `if` with the one before

File: services/ske/wait_test.go [hunk 3]
```diff
@@ -179,13 +200,14 @@ func TestCreateOrUpdateClusterWaitHandler(t *testing.T) {
...
-				resourceState: tt.resourceState,
+				getFails:             tt.getFails,
+				name:                 name,
+				resourceState:        tt.resourceState,
+				invalidArgusInstance: tt.invalidArgusInstance,
 			}
 			var wantRes *ClusterResponse
 			rs := ClusterStatusState(tt.resourceState)
-			if !tt.getFails {
+			if !tt.getFails && tt.resourceState != stateUnhealthy {
```
Comment (hcsa73) [inline services/ske/wait_test.go:210]: Doesn't this extra condition make the test not check the hander's response when `tt.invalidArgusInstance=true`? In that case you'd want to check if `gotRes` matches what you'd expect

File: services/ske/wait.go [hunk 4]
```diff
@@ -10,11 +10,13 @@ import (
...
-	StateHibernated = "STATE_HIBERNATED"
-	StateFailed     = "STATE_FAILED"
-	StateDeleting   = "STATE_DELETING"
-	stateCreated    = "STATE_CREATED"
+	StateHealthy                  = "STATE_HEALTHY"
+	StateHibernated               = "STATE_HIBERNATED"
+	StateFailed                   = "STATE_FAILED"
+	StateDeleting                 = "STATE_DELETING"
+	stateCreated                  = "STATE_CREATED"
+	stateUnhealthy                = "STATE_UNHEALTHY"
```
Comment (joaopalet) [inline services/ske/wait.go:18]: Maybe capitalise these two to match the others

File: services/ske/__debug_bin
Comment (hcsa73) [inline services/ske/__debug_bin:1]: Leftover file

**Review by hcsa73 — APPROVED** [review]

File: services/ske/__debug_bin
Comment (vicentepinto98) [reply services/ske/__debug_bin:1]: no :(((

**Review by hcsa73 — APPROVED** [review]

**Pull request merged**